### PR TITLE
[SMALLFIX] use fingerprint from ufsstatus if possible

### PR DESCRIPTION
**Pull request by yuzhu**
This will speed up list recursive.

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang  PTAL

**Comment by yuzhu** [comment]
jenkins, test this please

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -1198,12 +1198,17 @@ private void completeFileAndJournal(RpcContext rpcContext, LockedInodePath inode
...
       // Retrieve the UFS fingerprint for this file.
       MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
       AlluxioURI resolvedUri = resolution.getUri();
       try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
         UnderFileSystem ufs = ufsResource.get();
-        ufsFingerprint = ufs.getFingerprint(resolvedUri.toString());
+        if (ufsStatus == null) {
+          ufsFingerprint = ufs.getFingerprint(resolvedUri.toString());
+        } else {
+          ufsFingerprint = Fingerprint.create(ufs.getUnderFSType(), ufsStatus).serialize();
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: Do we have to do the equivalent for directories?

Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: added fingerprint creation on the createDirectory path

**Comment by yuzhu** [comment]
@gpang addressed comments, PTAL. thanks.

**Pull request merged**